imblearn/_min_dependencies.py

```diff
@@ -4,10 +4,10 @@
 NUMPY_MIN_VERSION = "1.17.3"
 SCIPY_MIN_VERSION = "1.3.2"
 PANDAS_MIN_VERSION = "1.0.5"
-SKLEARN_MIN_VERSION = "1.1.0"
+SKLEARN_MIN_VERSION = "1.1.3"
```

Comment:
Basically, in #946 I want to try to be more lenient and support 1.0.2.
I will first fix whatever else for 1.2 and try to get backward to see what do we need to port to be backward compatible. Right now this is a bit fuzzy.

Reply:
One complication is that 1.0.2 supports Python 3.7 ([end of life in ~6 months](https://endoflife.date/python)). If we stick with 3.8 as the minimum version we shouldn't have to mess with _min_versions too much.

Reply:
We can align our minimum Python with the NEP29. The idea would be to support scikit-learn 1.0.2 for the meaningful Python version.